manifests/auto_cloud_swarm.pp

```diff
@@ -0,0 +1,68 @@
+# == Class: robottelo_slave::auto_cloud_swarm
+#   Automatically setup everything needed to get a host from blank OS install up
+#   to a fully functioning slave registered to Jenkins with the swarm plugin
+#
+# === Parameters
+#
+# [*jenkins_master*]
+#   The jenkins master server to attach to
+#
+class robottelo_slave::auto_cloud_swarm(
+  $jenkins_master = undef,
+  $jenkins_user   = undef,
+  $jenkins_pwd    = undef,
+  $jenkins_labels = 'builders',
+  $slave_user     = 'jenkins',
+) {
+  yumrepo { 'epel-temp':
+    mirrorlist => 'https://mirrors.fedoraproject.org/metalink?repo=epel-7&arch=$basearch',
+    gpgcheck   => 0,
+    enabled    => 1,
+  } ->
+  package {'epel-release':
+    ensure => 'latest',
+    before => [
+      Package['mock', 'tito', 'git-annex', 'koji', 'npm'],
+      Class['robottelo_slave'],
+    ],
+  }
+
+  ensure_resource('group', $slave_user, { 'ensure' => 'present' })
+  ensure_resource('user', $slave_user, { 'ensure' => 'present' })
+  User<| title == $slave_user |> {
+    managehome => true,
+    gid        => $slave_user,
+    groups     => [$slave_user, 'mock'],
+    require    => Package['mock'],
+  }
+  User[$slave_user] -> Class['robottelo_slave']
+
+  include ::robottelo_slave::brew
+  ensure_packages(['mock'])
```

Comment:
ok, now I see what you are asking. you are right, ideally speaking we could
have 4 different possible configurations:
- "plain" slave
- brew enabled slave (so called "build" slave)
- automated "plain" slave
- automated brew enabled slave
right now the 3rd configuration is not possible to achieve. but it would
require some rethinking of apply.sh. so i suggest we leave it like this for
now to allow me to finish the work to get "builder" creation automated, and
create an issue to remind us to revisit this.
בתאריך 3 בפבר׳ 2016 05:22,‏ "Eric D Helms" <notifications@github.com> כתב:

> In manifests/auto_cloud_swarm.pp
> <https://github.com/SatelliteQE/puppet-robottelo_slave/pull/9#discussion_r51674236>
> :
>
> > +      Class['robottelo_slave'],
> > +    ],
> > +  }
> > +
> > +  ensure_resource('group', $slave_user, { 'ensure' => 'present' })
> > +  ensure_resource('user', $slave_user, { 'ensure' => 'present' })
> > +  User<| title == $slave_user |> {
> > +    managehome => true,
> > +    gid        => $slave_user,
> > +    groups     => [$slave_user, 'mock'],
> > +    require    => Package['mock'],
> > +  }
> > +  User[$slave_user] -> Class['robottelo_slave']
> > +
> > +  include ::robottelo_slave::brew
> > +  ensure_packages(['mock'])
>
> OK - I wasn't sure if this class should strictly handle configuring the
> jenkins slave aspect and the "builder" related packages etc. should be
> covered separately.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SatelliteQE/puppet-robottelo_slave/pull/9/files#r51674236>
> .
>